app/src/main/java/org/piramalswasthya/sakhi/ui/home_activity/non_communicable_diseases/ncd_referred/form/NcdRefferalFormViewModel.kt

```
@@ -0,0 +1,165 @@
+
+    fun saveDraft() {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                try {
+                    _state.postValue(State.SAVING)
+                    dataset.mapValues(referalCache)
+                    referalCache.syncState = SyncState.DRAFT
+                    // Assuming referalRepo has a save function that handles draft state
+                    // referalRepo.saveReferedNCD(referalCache)
+                    _state.postValue(State.DRAFT_SAVED)
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)